custom/icds_reports/management/commands/custom_data_pull_ap.py

```diff
@@ -0,0 +1,88 @@
+import csv
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+from django.core.management.base import BaseCommand
+from django.db import connections, transaction
+
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+
+query_to_fetch_awcs = """
+    SELECT doc_id, awc_name, awc_site_code, supervisor_name, block_name, district_name
+    FROM awc_location WHERE aggregation_level=5 AND state_id='f98e91aa003accb7b849a0f18ebd7039'
+"""
+
+#                                                               QUERY PLAN
+# ---------------------------------------------------------------------------------------------------------------------------------------
+#  Custom Scan (Citus Router)  (cost=0.00..0.00 rows=0 width=0)
+#    Task Count: 1
+#    Tasks Shown: All
+#    ->  Task
+#          Node: host=100.71.184.222 port=6432 dbname=icds_ucr
+#          ->  Index Scan using awc_location_pkey_102840 on awc_location_102840 awc_location  (cost=0.68..51589.67 rows=55002 width=109)
+#                Index Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
+#                Filter: (aggregation_level = 5)
+# (8 rows)
+
+query_to_fetch_visit_data = """
+    SELECT awc_id, CASE WHEN expected_visits>0 THEN valid_visits/(expected_visits)::float*100 ELSE 0 END as visits FROM agg_awc
```

Comment:
you also need to check if expected_visits is not 0 otherwise it will give division by 0 error

Reply:
It's already checking